### ISSUE TITLE
Class Diagram: User

### ISSUE THREAD

**Issue by egetaga** [bounswe/bounswe2022group1#94]
On our meeting with the team members, we decided the necessary qualifications of the **user** class diagram: 

- It should be an abstract class, since "user" has 2 child user types as "Guest" and  "Teacher".
- It should include "cookies", "ip" and "id".
- ip is the ip of the user and may change at each session. 
- id is the assigned session ID.
- and sign-in operation, since sign-in  is shared by all children types.

The user diagram should be dragged according to this specifications.

**Comment by egetaga** [comment]
According to the specifications decided by our team members, I created the user diagram. The fields are private in general and accessible only by means of getter and setter methods. 

- Since according to the users, every user shall be able to check the requirements, I have added check_course_content().
- The need for such a method is decided by all of our team members. 
- The class needs review by another team member. 
<img width="218" alt="Screen Shot 2022-04-05 at 11 21 12" src="https://user-images.githubusercontent.com/42147612/161711263-f144ec0f-5d0f-4966-a04e-9a767af37923.png">
